WcaOnRails/app/models/user.rb

```diff
@@ -579,7 +563,7 @@ def admin?
   end
 
   def any_kind_of_delegate?
-    delegate_status.present?
+    delegate_status.present? || active_roles.any? { |role| role.group_type == UserGroup.group_types[:delegate_regions] }
```

Comment:
active_roles of user object returns only those roles which are already migrated to roles table. So, it will currently cover only regional_delegates. Once senior_delegates are migrated to roles, this function will work without any change. Hopefully next time function needs to be touched is in the PR where we remove delegate_status from users table.